superset-frontend/src/SqlLab/components/ResultSet/ResultSet.test.tsx

```diff
@@ -224,14 +224,14 @@ describe('ResultSet', () => {
   });
 
   test('renders if there is no limit in query.results but has queryLimit', async () => {
-    const { getByRole } = setup(mockedProps, mockStore(initialState));
-    expect(getByRole('grid')).toBeInTheDocument();
+    const { getByTestId } = setup(mockedProps, mockStore(initialState));
+    expect(getByTestId('table-container')).toBeInTheDocument();
```

Comment:
It's important to follow [RTL Guiding principles](https://testing-library.com/docs/queries/about#priority) and use `getByTestId` as the last resort.

```suggestion
    const { getByRole } = setup(mockedProps, mockStore(initialState));
    expect(getByRole('table')).toBeInTheDocument();
```